src/compile/Component.ts

```diff
@@ -1116,7 +1116,9 @@ export default class Component {
 								const owner = scope.find_owner(name);
 								if (
 									(!owner || owner === component.instance_scope) &&
-									(name[0] === '$' || component.var_lookup.has(name) && component.var_lookup.get(name).writable)
+									(name[0] === '$' || component.var_lookup.has(name) &&
```

Comment:
Can this be split onto lines differently? Having the higher precedence `&&` split onto two lines, the first of which contains a lower precedence `||`, is confusing.

Reply:
I agree. I updated with a more readable alternative.